File: python/completed/euler0016.py
###
#
# Created by Holden on 12/17/2015
#
# SOLVED on 1/10/2016
#
# Problem 16 - Power Digit Sum
#
###

def power_digit_sum(exponent):
    
    value = 0
    
    for digit in str(2 ** exponent):
        value += int(digit)
        
    return value

File: java/completed/euler0016.java
import lib.EulerTime;

import java.math.BigInteger;

/**
 * Created by Holden on 12/02/2016. - COMPLETED
 *
 * Problem 16 - Power Digit Sum
 *
 */

public class Prob0016 {
    public static void main(String[] args) {
        long startTime = System.nanoTime();
        long answer = powerDigitSum(1000);
        long endTime = System.nanoTime();

        System.out.println(answer);

        long duration = (endTime - startTime);
        EulerTime.time(duration);
    }

    public static long powerDigitSum(int power) {
        BigInteger total = BigInteger.valueOf(1);
        BigInteger two = BigInteger.valueOf(2);
        for (int i = 0; i < power; i++) {
            total = total.multiply(two);
        }

        int digitSum = 0;
        char[] digits = total.toString().toCharArray();
        for (char i: digits) {
            digitSum += Character.getNumericValue(i);
        }

        return digitSum;
    }
}
